doc/influxql-2.md

```diff
@@ -0,0 +1,910 @@
+* **database** - an object with a name and a retention [duration string](https://golang.org/pkg/time/#ParseDuration). The duration determines how long data is kept in the database. It is a container of raw series. Names must be a string and can contain `_`, `-`, `0-9`, `a-z`, `A-Z`. Database names MUST not start with `_`. Note that retention policies are no longer a part of the model. Example:
+
+```json
+{
+	"name": "foo",
+	"retention": "0"
+}
+```
+
+### Proposed Write Protocol
+
+The primary write protocol is a line protocol. In the context of a write, only **raw points** can be written. The line protocol contains some shortcuts to avoid the duplication of shared information like tags and time, but ultimately those will be converted into raw points, which are written into raw series (meaning that points in a series MUST have a unique time or they will overwrite previously written points).
+
+The protocol can also include comments, which are lines that start with `#` followed by a space. Lines that start with `##` specify a block of meta data for the line immediately following.
+
+The most basic style of the protocol has the tag key/value pairs separated by `:` with a trailing space, a value, and a trailing epoch or time. The structure looks like this:
+
+```
+<tagset> <value> <time>
+```
+
+All three sections are required. Here are the rules for each section.
+
+#### Tagset
+
+The tagset is the key/value pairs that identify the series. A line MUST have at least one tag key/value pair. They should be separated by `:`. Spaces, commas, colons, and backslashes in either a key or value MUST be escaped by a backslash.
+
+#### Value
+
+The value block can be any of the value types: boolean, float64, int64, uint64, time, string, or bytes. Null values should never be specified as they aren't stored in the database. Boolean is represented with `t` or `f`. Int64, UInt64 must have a trailing `i` or `u`, while all numbers without the trailing character are parsed as float64. Times must be in RFC3339Nano format. Strings should be wrapped in double quotes, which means any double quote within a string must be escaped with a backslash. Bytes look like an array of numbers: `[102, 111, 111]`
+
+#### Time
+
+The time block can be specified as either an epoch or an RFC3339Nano string. If an epoch is used it should be followed by its precision. However, if no precision is specified, the system will guess the precision by whatever is closest to the current time on the server receiving the write.
+
+#### Meta data
+
+*TODO:* give this more definition.
+
+Meta data can be written with a line starting with `##`. The meta data will be associated with the series on the next line. Meta data is a JSON object. Keys in the object starting with `_` are reserved for the system. The data in meta is not indexed. The only validation is that it is a parseable JSON object.
+
```

Comment:
You'd want to use tags for that, since those would be attributes you'd be likely to query by, so I'd be looking at either making them smart enough to send those tags natively or using an "enrichment" layer in to add them as part of the ingest pipeline.